$("#helpref").html("1. Zubradt M, Gupta P, Persad S, et al. DMS-MaPseq for genome-wide or targeted RNA structure probing in vivo. Nat. Methods 2017; 14:75–82<br />2. Wang Z, Ma Z, Castillo-Gonzalez C, et al. SWI2/SNF2 ATPase CHR2 remodels pri-miRNAs via Serrate to impede miRNA production. Nature 2018; 557:516–521<br />3. Guenther UP, Weinberg DE, Zubradt MM, et al. The helicase Ded1p controls use of near-cognate translation initiation codons in 5’ UTRs. Nature 2018; 559:130–134<br />4. Incarnato D, Morandi E, Simon LM, et al. RNA Framework: an all-in-one toolkit for the analysis of RNA structures and post-transcriptional modifications. Nucleic Acids Res. 2018; 46:e97<br />5. Burkhardt DH, Rouskin S, Zhang Y, et al. Operon mRNAs are organized into ORF-centric structures that predict translation efficiency. Elife 2017; 6:<br />6. Ritchey LE, Su Z, Tang Y, et al. Structure-seq2: sensitive and accurate genome-wide profiling of RNA structure in vivo. Nucleic Acids Res. 2017; 45:e135<br />7. Incarnato D, Neri F, Anselmi F, et al. Genome-wide profiling of mouse RNA secondary structures reveals key features of the mammalian transcriptome. Genome Biol 2014; 15:491<br />8. Ding Y, Tang Y, Kwok CK, et al. In vivo genome-wide profiling of RNA secondary structure reveals novel regulatory features. Nature 2014; 505:696–700<br />9. Rouskin S, Zubradt M, Washietl S, et al. Genome-wide probing of RNA structure reveals active unfolding of mRNA structures in vivo. Nature 2014; 505:701–705<br />10. Beaudoin JD, Novoa EM, Vejnar CE, et al. Analyses of mRNA structure dynamics identify embryonic gene regulatory programs. Nat. Struct. Mol. Biol. 2018; 25:677–686");